src/components/TextField.vue

```diff
@@ -183,6 +183,15 @@ export default {
         'text-field--minimized-label': minimizedLabel,
         [`text-field--${styleVariant}`]: styleVariant !== '',
       };
+
+      /**
+       * If the input is a currency field, allow 0 to show in the modified state.
+       */
+      if (currency && value === 0) {
+        classes['text-field--modified'] = true;
+      }
```

Comment:
I don't understand this logic - why do you need to mark the field as modified if it has a value of 0?

Reply:
Because technically it is modified in this case. The value is just zero. It needs to be marked as Modified so that the Currency symbol is visible as well as any other styling that's inherited by an input being marked as such.  Hopefully, that makes sense haha

Reply:
What I'm reading is that if I render `<TextField value="0" currency />` in a template, it will have the `text-field--modified` class added to it before I've modified anything - that feels odd

Reply:
Hmm okay, I can see how that feels odd. I was looking at it not from a user modifying it but the input being modified (in this case by the value being present). I'll see if I can break it out to make it a bit cleaner.

Reply:
I think we should drop "modified" as we already have "empty". It seems like "modified" is a badly named opposite to "empty".